src/Behat/Config/Formatter/Formatter.php

```diff
@@ -59,4 +82,65 @@ public static function defaults(): array
     {
         return [];
     }
+
+    /**
+     * @internal
+     */
+    public function toPhpExpr(): Expr
+    {
+        $formatterObject =  $this->builderFactory->new(new FullyQualified(self::class));
+
+        $expr = $this->applyBaseSettings($formatterObject);
+
+        if ($this->settings === []) {
+            $args = $this->builderFactory->args([$this->name]);
+        } else {
+            $args = $this->builderFactory->args([$this->name, $this->settings]);
+        }
+        $formatterObject->args = $args;
+
+        return $expr;
+    }
+
+    /**
+     * @internal
+     */
+    protected function toPhpExprForNamedFormatter(): Expr
+    {
+        $formatterObject =  $this->builderFactory->new(new FullyQualified(static::class));
+
+        $expr = $this->applyBaseSettings($formatterObject);
+
+        $defaults = static::defaults();
+        $argValues = [];
+        foreach ($defaults as $name => $defaultValue) {
+            if ($this->settings[$name] !== $defaultValue) {
+                $value = $this->settings[$name];
+                if ($name === ShowOutputOption::OPTION_NAME) {
+                    $value = ShowOutputOption::from($value);
+                    $name = ShowOutputOption::PARAMETER_NAME;
+                }
+                $argValues[$name] = $value;
+            }
+        }
```

Comment:
It's a personal preference, but here I would also use an early return to avoid nested conditions.

```suggestion
        foreach ($defaults as $name => $defaultValue) {
            if ($this->settings[$name] === $defaultValue) {
                continue;
            }
            
            $value = $this->settings[$name];
            if ($name === ShowOutputOption::OPTION_NAME) {
                $value = ShowOutputOption::from($value);
                $name = ShowOutputOption::PARAMETER_NAME;
            }
            
            $argValues[$name] = $value;
        }
```

Reply:
Nice one, updated